docs/guide/custom-functions.md

```diff
@@ -119,72 +125,116 @@ Register your function plugin (and its translations) so that HyperFormula can re
 
-You can configure your custom function to:
-* Use the [array arithmetic mode](arrays.md).
-* Treat reference or range arguments as arguments that don't create dependency.
-* Inline range arguments to scalar arguments.
-* Get recalculated with each sheet shape change.
-* Be a [volatile](volatile-functions.md) function.
-* Repeat a specified number of last arguments indefinitely.
-* Never get vectorized.
+You can set the following options for your function:
+
+| Option                              | Type    | Description                                                                                                                                  |
+|-------------------------------------|---------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| `method` (required)                 | String  | Name of the method that implements the custom function logic.                                                                                |
+| `parameters`                        | Array   | Specification of the arguments accepted by the function and their [validation options](#argument-validation-options).                        |
+| `repeatLastArgs`                    | Number  | For functions with a variable number of arguments: sets how many last arguments can be repeated indefinitely.                                |
+| `expandRanges`                      | Boolean | `true`: ranges in the function's arguments are inlined to (possibly multiple) scalar arguments.                                              |
+| `returnNumberType`                  | String  | Function return type.                                                                                                                        |
+| `arraySizeMethod`                   | ?       | ?                                                                                                                                            |
+| `isVolatile`                        | Boolean | `true`: the function is [volatile](volatile-functions.md).                                                                                   |
+| `isDependentOnSheetStructureChange` | Boolean | `true`: the function gets recalculated with each sheet shape change (e.g., when adding/removing rows or columns).                            |
+| `doesNotNeedArgumentsToBeComputed`  | Boolean | `true`: the function treats reference or range arguments as arguments that don't create dependency (other arguments are properly evaluated). |
+| `arrayFunction`                     | Boolean | `true`: the function enables the [array arithmetic mode](arrays.md) in its arguments and nested expressions.                                 |
+| `vectorizationForbidden`            | Boolean | `true`: the function will never get vectorized.                                                                                              |
```

Comment:
Can you explain what it means to vectorize a function? This concept is not present in the docs.

Reply:
https://hyperformula.handsontable.com/guide/arrays.html#passing-arrays-to-scalar-functions

Reply:
But it's hard to find indeed (the vuepress search looks only at the headlines). We might rephrase that a little and link that section to the custom-functions guide. WDYT?

Reply:
Yes, that would be fine.

Reply:
Done